[GTK][WPE] Remote Inspector: speed up of reading large message

pgorszkowski-igalia opened the pull request:
#### 58384db7507ff0632dd6d53d57fb80b72b3c2162
<pre>
[GTK][WPE] Remote Inspector: speed up of reading large message
<a href="https://bugs.webkit.org/show_bug.cgi?id=285978">https://bugs.webkit.org/show_bug.cgi?id=285978</a>

Reviewed by Adrian Perez de Castro.

In case of large message, we can speed its reading up by increase the reading buffer to
expected size of the message (which is read from the header of the message).

The solution was proposed in <a href="https://github.com/WebPlatformForEmbedded/WPEWebKit/pull/1407.">https://github.com/WebPlatformForEmbedded/WPEWebKit/pull/1407.</a>

Original author: Krishna Priya K (<a href="https://github.com/kkanag314).">https://github.com/kkanag314).</a>

* Source/WTF/wtf/glib/SocketConnection.cpp:
(WTF::SocketConnection::readMessage):

Canonical link: <a href="https://commits.webkit.org/289009@main">https://commits.webkit.org/289009@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/0ffb4fc9ad6e094bc4c36539053b4ab42d33bf70

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/84810 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/131/builds/4535 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/55/builds/39198 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/89949 "Built successfully") | [✅ 🛠 win](https://ews-build.webkit.org/#/builders/59/builds/35862 "Built successfully") 
| | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/130/builds/4624 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/123/builds/12510 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/65991 "Passed tests") | [✅ 🧪 win-tests](https://ews-build.webkit.org/#/builders/60/builds/23813 "Passed tests") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/87855 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/132/builds/3518 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/77065 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/46278 "Passed tests") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/133/builds/3397 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/64/builds/31287 "Passed tests") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/34936 "Built successfully") | 
| [✅ 🛠 🧪 jsc](https://ews-build.webkit.org/#/builders/20/builds/77773 "Built successfully and passed tests") | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/74261 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/63/builds/32096 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/91325 "Built successfully") | 
| [✅ 🛠 🧪 jsc-arm64](https://ews-build.webkit.org/#/builders/12/builds/84028 "Built successfully and passed tests") | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/128/builds/12149 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/122/builds/8865 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/74473 "Passed tests") | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/121/builds/12376 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/72879 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/73598 "Passed tests") | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/18258 "Built successfully and passed tests") | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/126/builds/17989 "Passed tests") | [✅ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/119/builds/16422 "Passed tests") | [✅ 🛠 playstation](https://ews-build.webkit.org/#/builders/134/builds/3597 "Built successfully") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/127/builds/12101 "Built successfully") | [✅ 🛠 mac-safer-cpp](https://ews-build.webkit.org/#/builders/120/builds/17541 "Built successfully") | [✅ 🛠 jsc-armv7](https://ews-build.webkit.org/#/builders/35/builds/106245 "Built successfully") | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/125/builds/11935 "Built successfully") | | [✅ 🧪 jsc-armv7-tests](https://ews-build.webkit.org/#/builders/25/builds/25643 "Passed tests") | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/129/builds/15429 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/124/builds/13681 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

aperezdc reviewed (approved):
LGTM. The data for each message needs to be read in memory anyway in order to deserialize a `GVariant` so it makes sense to preallocate space in the buffer instead of iterating a few times to add space incrementally.

FWIW, I can think of a few ways to improve the surrounding code, but that would be a different patch. Let's merge this as-is for now 😄

pgorszkowski-igalia committed:
[GTK][WPE] Remote Inspector: speed up of reading large message
https://bugs.webkit.org/show_bug.cgi?id=285978

Reviewed by Adrian Perez de Castro.

In case of large message, we can speed its reading up by increase the reading buffer to
expected size of the message (which is read from the header of the message).

The solution was proposed in https://github.com/WebPlatformForEmbedded/WPEWebKit/pull/1407.

Original author: Krishna Priya K (https://github.com/kkanag314).

* Source/WTF/wtf/glib/SocketConnection.cpp:
(WTF::SocketConnection::readMessage):

Canonical link: https://commits.webkit.org/289009@main
